src/core/load-fragment.c

```diff
@@ -4629,6 +4632,49 @@ int config_parse_load_credential(
                 }
         }
 
+        r = stat(q, &st);
+        if (r < 0) {
+                log_unit_error_errno(u, errno, "Couldn't stat credential at %s", q);
+                return 0;
+        }
+
+        if (st.st_mode & S_IFDIR) {
+                _cleanup_closedir_ DIR *dir = NULL;
+                struct dirent *dent;
+
+                dir = opendir(q);
+                if (!dir) {
+                        log_unit_error_errno(u, errno, "Couldn't open directory %s", q);
+                        return 0;
+                }
+
+                log_unit_info(u, "Loading credentials from directory %s", q);
+                FOREACH_DIRENT(dent, dir, return -errno) {
+                        _cleanup_free_ char *new_rvalue = NULL, *new_path = NULL;
+
+                        /* avoid infinite recursion */
+                        if (dent->d_type == DT_LNK) {
```

Comment:
this is not sufficient, as d_type is an optional field and might be DT_UNKNOWN

moreover this check is racy and should be inverted, to only allow DT_REG